File: structures-core/src/main/java/org/kinotic/structures/api/domain/Project.java
package org.kinotic.structures.api.domain;

import java.util.Date;
import org.kinotic.continuum.api.Identifiable;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

@Getter
@Setter
@Accessors(chain = true)
@NoArgsConstructor
public class Project implements Identifiable<String> {

    /**
     * The id of the project.
     * All project ids are unique throughout the entire system.
     */
    private String id;

    /**
     * The id of the application that this project belongs to.
     * All application ids are unique throughout the entire system.
     */
    private String applicationId;
    
    /**
     * The name of the project.
     */
    private String name;

    /**
     * The description of the project.
     */
    private String description;

    /**
     * The source of truth for the project.
     */
    private ProjectType sourceOfTruth;

    /**
     * The date and time the project was updated.
     */
    private Date updated;

}

File: structures-js/structures-api/src/api/domain/Project.ts
import { Identifiable } from '@kinotic/continuum-client'
import { ProjectType } from '@/api/domain/ProjectType'

export class Project implements Identifiable<string> {
    /**
     * The id of the project.
     * All project ids are unique throughout the entire system.
     */
    public id: string | null = null

    /**
     * The id of the application that this project belongs to.
     * All application ids are unique throughout the entire system.
     */
    public applicationId: string

    /**
     * The name of the project.
     */
    public name: string

    /**
     * The description of the project.
     */
    public description?: string

    /**
     * The source of truth for the project.
     */
    public sourceOfTruth: ProjectType | null = null

    /**
     * The date and time the project was updated.
     */
    public updated: number | null = null

    constructor(id: string | null, applicationId: string, name: string, description?: string) {
        this.id = id
        this.applicationId = applicationId
        this.name = name
        this.description = description
    }
}
